core/blockchain.go

```diff
@@ -1168,8 +1176,33 @@ func (bc *BlockChain) WriteBlockWithState(
 
 	// Flush trie state into disk if it's archival node or the block is epoch block
 	triedb := bc.stateCache.TrieDB()
+
 	if bc.cacheConfig.Disabled || block.IsLastBlockInEpoch() {
-		if err := triedb.Commit(root, false); err != nil {
+		if bc.cacheConfig.Disabled && !block.IsLastBlockInEpoch() {
```

Comment:
We still need to flush the state into disk for archival node, except for validatorWrapper data. Other state changes still need to be flushed to disk.

Reply:
I see, the other states need to be committed every block, while the ValidatorWrapper does not need to be committed per block, but keeps the current commit every N blocks(it still commit finally, It only reduces the frequency of its commit). right ?

Reply:
Yes